src/Access/ContextAccess.cpp

```diff
@@ -320,6 +320,20 @@ std::shared_ptr<const AccessRights> ContextAccess::getAccessRightsWithImplicit()
     return access_with_implicit;
 }
 
+Strings ContextAccess::getActiveProfileNames() const
+{
+    std::lock_guard lock{mutex};
+    Strings result;
+    if (!enabled_settings)
+        return result;
+
+    const auto profiles_ids = enabled_settings->getAppliedProfiles();
+    result.reserve(profiles_ids.size());
+    for (const auto & profile_id : profiles_ids)
+        result.push_back(manager->readName(profile_id));
```

Comment:
`SettingsProfileCache` keeps a map named `all_profiles`, which can be used to get all those names. Searching in the map is faster than the call `manager->readName()`.

Reply:
You can easily implement this by adding a new member variable of type `const SettingsProfileCache &` to `EnabledSettings` and also the function `EnabledSettings::getCurrentProfileNames()`.

Reply:
done, in a bit different way